GVRf/Framework/src/org/gearvrf/GVRContext.java

```diff
@@ -417,6 +427,172 @@ public void loadMesh(MeshCallback callback, GVRAndroidResource resource,
     }
 
     /**
+     * Simple, high-level method to load a scene as {@link GVRSceneObject} from
+     * 3D model.
+     * 
+     * @param assetRelativeFilename
+     *            A filename, relative to the {@code assets} directory. The file
+     *            can be in a sub-directory of the {@code assets} directory:
+     *            {@code "foo/bar.png"} will open the file
+     *            {@code assets/foo/bar.png}
+     * 
+     * @return A {@link GVRSceneObject} that contains the meshes with textures
+     * 
+     * @throws IOException
+     *             File does not exist or cannot be read
+     */
+    public GVRSceneObject getAssimpModel(String assetRelativeFilename)
+            throws IOException {
+        GVRAssimpImporter assimpImporter = GVRImporter.readFileFromResources(
+                this, new GVRAndroidResource(this, assetRelativeFilename));
+
+        GVRSceneObject wholeSceneObject = new GVRSceneObject(this);
+
+        AiScene assimpScene = assimpImporter.getAssimpScene();
+
+        AiWrapperProvider<byte[], AiMatrix4f, byte[], AiNode, byte[]> wrapperProvider = new AiWrapperProvider<byte[], AiMatrix4f, byte[], AiNode, byte[]>() {
+
+            @Override
+            public byte[] wrapColor(ByteBuffer arg0, int arg1) {
+                byte[] colorBuffer = new byte[4];
+                arg0.get(colorBuffer, arg1, 4);
+                return colorBuffer;
+            }
+
+            @Override
+            public AiMatrix4f wrapMatrix4f(float[] arg0) {
+
+                AiMatrix4f transformMatrix = new AiMatrix4f(arg0);
+                return transformMatrix;
+            }
+
+            @Override
+            public byte[] wrapQuaternion(ByteBuffer arg0, int arg1) {
+                byte[] quaternion = new byte[4];
+                arg0.get(quaternion, arg1, 4);
+                return quaternion;
+            }
+
+            @Override
+            public AiNode wrapSceneNode(Object arg0, Object arg1, int[] arg2,
+                    String arg3) {
+
+                AiNode node = new AiNode(null, arg1, arg2, arg3);
+
+                return node;
+            }
+
+            @Override
+            public byte[] wrapVector3f(ByteBuffer arg0, int arg1, int arg2) {
+                byte[] warpedVector = new byte[arg2];
+                arg0.get(warpedVector, arg1, arg2);
+                return warpedVector;
+            }
+        };
+
```

Comment:
The argX seems to be a bad naming way to me. It is very hard to understand what it is doing. I think you'd better change them to better namings which can represent what they are.